### ISSUE TITLE
Migrate away from deprecated Platform/Board Configurations

### ISSUE THREAD

**Issue by barndawgie** [esphome-econet/esphome-econet#323]
Per [current ESPHome documentation](https://esphome.io/components/esphome), the `Platform` and `Board` settings under `esphome` are deprecated.

@bitfliq indicated this is preventing him from making some desired configuration updates: https://discord.com/channels/1148015790038188073/1148015790520541246/1196697510840897547

**Comment by barndawgie** [comment]
At first glance, this isn't a super-hard change to make, but it may not be possible without breaking existing configurations.

**Comment by segdy** [comment]
Yes, I have exactly the issue right now that I need to tune items (framework type) but can't because of the old format.

**Comment by segdy** [comment]
Created a pull request: https://github.com/esphome-econet/esphome-econet/pull/422

Hopefully it makes it in, otherwise it's impossible to use Atom S3 Lite for example
